### PR TITLE
Add option for not trimming input value

### PR DESCRIPTION
**Pull request by LukaszKokot**
We add a "trim" option that is by default set to "true", in order for the
component to stay backward compatible. However, when this property is set
to "false", the value is the one directly entered by the user, with all
whitespaces left untouched.

### CODE REVIEW

**Review by RichardCzechowski — APPROVED** [review]
This looks good to me. I'm not overly familiar with this code, but from what I do know, I'd be happy bringing it in and verifying in Looker.
Looks like you've essentially built https://github.com/angular-ui/ui-select/issues/2064, we should discuss whether this is something we should open back into the original repo.

File: test/select.spec.js [hunk 1]
```diff
@@ -1139,7 +1140,7 @@ describe('ui-select tests', function () {
       var el = createUiSelect();
       expect(el.find('.ui-select-choices-group .ui-select-choices-group-label').map(function () {
         return this.textContent;
-      }).toArray()).toEqual(["Foo", "Baz", "bar"]);
```
Comment (rlm1023) [inline test/select.spec.js:1142]: Interesting that this line changed.

**Review by rlm1023 — APPROVED** [review]
👍 . So I didn't even know we had this repo. Might want someone that's worked in here before to review.

**Comment by rlm1023** [comment]
The last few commits are me 🤦‍♂ . This is the angular ui-select, got it. Disregard my last comment.

**Comment by LukaszKokot** [comment]
It was really a quick fix, no worries @rlm1023. Thanks for the review!

**Comment by LukaszKokot** [comment]
@RichardCzechowski Looks like there already is a PR (cf. https://github.com/angular-ui/ui-select/pull/2157) open since August 2018. Not sure the original repo is really maintained anymore.

**Comment by rlm1023** [comment]
We really might want someone like @stalbot 's opinion here. A little worried with how this could fallout.

**Comment by rlm1023** [comment]
Who's on paternity leave. Maybe @lameyer ?

**Comment by lameyer** [comment]
uh oh I've been summoned 😂 I'll take a look

**Comment by LukaszKokot** [comment]
@rlm1023 @lameyer @RichardCzechowski For the record, I have a PR using this version of `ui-select` and the CI tests just turned all green. And since the trimming is being done by default (so in all other cases where the component is used), it's kinda safe I would say.

File: test/select.spec.js (hunk 1 above)
Comment (lameyer) [reply test/select.spec.js:1142]: Yeah that is weird... @LukaszKokot do you know why this changed?

**Review by lameyer — APPROVED** [review]
Since this is just adding an option, and by default does the exact same thing, I feel pretty comfortable with this. @LukaszKokot if you can confirm with manual testing when you have helltool with this change the other input boxes will still trim spaces, then I feel like we're good

**Comment by LukaszKokot** [comment]
@lameyer this is for fixing a unit test failing from the last commit to master. Please see Rob's comment, cf. https://github.com/looker/ui-select/pull/19#issuecomment-575675991 :-)

**Comment by rlm1023** [comment]
Sorry my "disregard comment" comment was about this one:

> So I didn't even know we had this repo. Might want someone that's worked in here before to review.

Not that unit test changing. Still want to make sure we understand why that changed.

**Comment by LukaszKokot** [comment]
So, I ran the tests on master, and that test I fixed was there, so something was wrong before my PR. And the last `master` commit is the one currently used in the main app code. Should I investigate further? (asking because I'm multi-tasking different issues at once)

**Comment by lameyer** [comment]
So this test was broken on master as well you're saying? That's probably fine then 🤷‍♀

**Pull request merged**